src/components/CollectionsPage/CollectionsPage.tsx

```diff
@@ -143,83 +145,90 @@ export default class CollectionsPage extends React.PureComponent<Props> {
     this.setState({ page: +props.activePage! }, this.isCollectionTabActive() ? this.fetchCollections : this.fetchItems)
   }
 
+  renderRightActions = () => {
+    const { view, isThirdPartyManager, onSetView } = this.props
+    return (
+      <Column align="right">
+        <Row className="actions">
+          {this.isCollectionTabActive() && (
+            <Dropdown
+              trigger={
+                <Button basic className="create-item">
+                  <Icon name="add-active" />
+                </Button>
+              }
+              inline
+              direction="left"
+            >
+              <Dropdown.Menu>
+                <>
+                  <Dropdown.Item text={t('collections_page.new_collection')} onClick={this.handleNewCollection} />
+                  {isThirdPartyManager && (
+                    <Dropdown.Item text={t('collections_page.new_third_party_collection')} onClick={this.handleNewThirdPartyCollection} />
+                  )}
+                </>
+              </Dropdown.Menu>
+            </Dropdown>
+          )}
+          <Button className="open-editor" primary onClick={this.handleOpenEditor} size="tiny">
+            {t('item_editor.open')}
+          </Button>
+          <Chip
+            className="grid"
+            icon="grid"
+            isActive={view === CollectionPageView.GRID}
+            onClick={() => onSetView(CollectionPageView.GRID)}
+          />
+          <Chip
+            className="list"
+            icon="table"
+            isActive={view === CollectionPageView.LIST}
+            onClick={() => onSetView(CollectionPageView.LIST)}
+          />
+        </Row>
+      </Column>
+    )
+  }
+
   renderPage() {
-    const { collectionsPaginationData, itemsPaginationData, view, isThirdPartyManager, onSetView, isLoadingItems, isLoadingCollections } =
-      this.props
+    const { collectionsPaginationData, itemsPaginationData, view, isLoadingItems, isLoadingCollections, hasUserOrphanItems } = this.props
     const { page } = this.state
     const totalCollections = collectionsPaginationData?.total
     const totalItems = itemsPaginationData?.total
     const count = this.isCollectionTabActive() ? totalCollections : totalItems
     const totalPages = this.isCollectionTabActive() ? collectionsPaginationData?.totalPages : itemsPaginationData?.totalPages
 
+    if (isLoadingItems || isLoadingCollections || count === undefined) {
```

Comment:
Should the loader wait for the UI to load if it has orphan items?

Reply:
good catch! it should wait